src/domain/infected-people/age-group/age-group-chart.tsx

```diff
@@ -0,0 +1,282 @@
+import css from '@styled-system/css';
+import { AxisBottom, TickRendererProps } from '@visx/axis';
+import { localPoint } from '@visx/event';
+import { GridColumns } from '@visx/grid';
+import { Group } from '@visx/group';
+import { scaleBand, scaleLinear } from '@visx/scale';
+import { Bar } from '@visx/shape';
+import { Text } from '@visx/text';
+import {
+  NationalInfectedAgeGroups,
+  NationalInfectedAgeGroupsValue,
+} from '~/types/data.d';
+import { formatPercentage } from '~/utils/formatNumber';
+import siteText from '~/locale/index';
+import { MouseEvent } from 'react';
+
+export const AGE_GROUP_TOOLTIP_WIDTH = 340;
+
+const text = siteText.infected_age_groups;
+
+interface AgeGroupChartProps {
+  parentWidth: number;
+  data: NationalInfectedAgeGroups;
+  openTooltip: (
+    event: MouseEvent<SVGGElement>,
+    value: NationalInfectedAgeGroupsValue,
+    getTooltipCoordinates: (
+      event: MouseEvent<SVGGElement>,
+      value: NationalInfectedAgeGroupsValue
+    ) => { x: number; y: number }
+  ) => void;
+  closeTooltip: () => void;
+}
+
+const TickValue = ({ x, y, formattedValue }: TickRendererProps) => {
+  return (
+    <Text x={x} y={y} fill="#595959" fontSize={'1rem'} textAnchor="middle">
+      {formattedValue}
+    </Text>
+  );
+};
+
+export const formatAgeGroupRange = (range: string): string => {
+  return range.split('-').join(' – ');
+};
+
+export function AgeGroupChart({
+  data,
+  parentWidth,
+  openTooltip,
+  closeTooltip,
+}: AgeGroupChartProps) {
+  const values = data.values.sort((a, b) => {
+    const aStart = parseInt(a.age_group_range, 10);
+    const bStart = parseInt(b.age_group_range, 10);
+    return bStart - aStart;
+  });
+
+  // Define the graph dimensions and margins
+  const isSmallScreen = parentWidth < 500;
+  const ageRangeAxisWidth = isSmallScreen ? 60 : 100;
+  const width = parentWidth;
+  const height = 400;
+  const marginX = isSmallScreen ? 6 : 40;
+  const margin = { top: 35, bottom: 20, left: marginX, right: marginX };
+
+  const numTicks = isSmallScreen ? 3 : 4;
+
+  // Then we'll create some bounds
+  const xMax = (width - margin.left - margin.right - ageRangeAxisWidth) / 2;
+  const yMax = height - margin.top - margin.bottom;
+
+  // We'll make some helpers to get at the data we want
+  const ageGroupPercentage = (d: NationalInfectedAgeGroupsValue) =>
+    d.age_group_percentage;
+  const infectedPercentage = (d: NationalInfectedAgeGroupsValue) =>
+    d.infected_percentage;
+  const y = (d: NationalInfectedAgeGroupsValue) => d.age_group_range;
```

Comment:
```suggestion
  const yDomain = (d: NationalInfectedAgeGroupsValue) => d.age_group_range;
```
Just for clarity sake?

Reply:
I'll rename this to `ageGroupRange` then, `yDomain` is also not correct/clear.

Reply:
nit: I must say that I like `x` and `y` since it maps to the axes

Reply:
Then call is xAxis and yAxis ;)

Reply:
yeah or getX, getY